File: bridgeserver.py
from BridgePython.bridge import Bridge 
bridge = Bridge(api_key='f99ede2465de4114')


def message_cb(msg):
  print msg

forjs = """
handler.run  =  function(message_cb, start, end) {
        var temp = start + end;
        console.log(temp);
        message_cb(temp);
}
"""

channel = bridge.get_channel('function-channel')
channel.message(forjs)

cluster = bridge.get_service('cluster')

cluster.run(message_cb, 1, 3)
cluster.run(message_cb, 5, 3)

bridge.connect()

File: Hadoop.js/bridgeserver.js
var bridge = new Bridge({apiKey: 'f99ede2465de4114'});

var funcReady = false;
var connectionsToWaitFor = 1;
var funcTask = '';

var numConnections = 0;
var handleConnection = {
	message: function(msg) {
		console.log("Connection received")
		numConnections++;
		if (numConnections === connectionsToWaitFor && funcReady) {
      funcReady = false;
			bridge.getChannel('function-channel', function(channel) {
			   channel.message(funcTask);
			});
		}
	}
}

bridge.joinChannel('connections', handleConnection);

var counter = 0;

var functionExecutor = {
  message: function(msg) {
    counter++;
    console.log(counter);
    if (counter === numConnections) {
      for(var i = 0; i < numConnections; i++) {
        bridge.getService('cluster', function(clusterService, name) {
          clusterService.run(connectionsToWaitFor, messageCB);
        });
      }
    }
  }
}

bridge.joinChannel('ack-channel', functionExecutor);
bridge.connect();
